src/ui/components/tenants/tenantDetail/thirdPartyPage/ThirdPartyPage.tsx

```diff
@@ -0,0 +1,436 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { useGetThirdPartyProviderInfo } from "../../../../../api/tenants";
+import { ProviderConfigResponse, TenantDashboardView } from "../../../../../api/tenants/types";
+import { getImageUrl, isValidHttpUrl } from "../../../../../utils";
+import Button from "../../../button";
+import { Loader } from "../../../loader/Loader";
+import { useTenantDetailContext } from "../TenantDetailContext";
+import { TenantDetailHeader } from "../TenantDetailHeader";
+import { PanelHeader, PanelHeaderTitleWithTooltip, PanelRoot } from "../tenantDetailPanel/TenantDetailPanel";
+import { ProviderInfoForm } from "../thirdPartyProviderConfig/ProviderInfoForm";
+import { ThirdPartyProviderInput } from "../thirdPartyProviderInput/ThirdPartyProviderInput";
+import "./thirdPartyPage.scss";
+
+export const ThirdPartyPage = ({
+	viewObj,
+	setViewObj,
+}: {
+	viewObj: TenantDashboardView;
+	setViewObj: Dispatch<SetStateAction<TenantDashboardView>>;
+}) => {
+	const handleProviderInfoBack = () => {
+		setViewObj({ view: "tenant-detail" });
+	};
+	return (
+		<div className="third-party-section">
+			<button
+				className="button flat"
+				onClick={() => handleProviderInfoBack()}>
+				<img
+					src={getImageUrl("left-arrow-dark.svg")}
+					alt="Go back"
+				/>
+				<span>Back to tenant info</span>
+			</button>
+			<div className="third-party-section__cards">
+				<TenantDetailHeader onlyShowTenantId />
+				{viewObj.view === "add-or-edit-third-party-provider" && (
```

Comment:
No it can never be false, so don't need the assert, since this component is only rendered by parent component if this condition is true, so don't need to check it again here, removed this condition